pkg/entrypoint/admissioncontroller/admissioncontroller_test.go

```diff
@@ -0,0 +1,1084 @@
+package admissioncontroller
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"reflect"
+	"regexp"
+	"testing"
+
+	_ "github.com/openshift/origin/pkg/api/install"
+	_ "github.com/openshift/origin/pkg/security/apis/security/install"
+	"github.com/sirupsen/logrus"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/fake"
+)
+
+type fakeResponseWriter struct {
+	statusCode int
+	h          http.Header
+	bytes.Buffer
+}
+
+func newFakeResponseWriter() *fakeResponseWriter {
+	return &fakeResponseWriter{
+		h:          map[string][]string{},
+		statusCode: http.StatusOK,
+	}
+}
+
+func (w *fakeResponseWriter) Header() http.Header {
+	return w.h
+}
+
+func (w *fakeResponseWriter) WriteHeader(statusCode int) {
+	w.statusCode = statusCode
+}
+
+func (w *fakeResponseWriter) Dump() {
+	fmt.Printf("HTTP %d %s\r\n", w.statusCode, http.StatusText(w.statusCode))
+	w.h.Write(os.Stdout)
+	fmt.Print("\r\n")
+	os.Stdout.Write(w.Bytes())
+}
+
+func TestHandleMalformedRequests(t *testing.T) {
+	client := fake.NewSimpleClientset(&core.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+			Annotations: map[string]string{
+				"openshift.io/sa.scc.uid-range": "1000/10",
+				"openshift.io/sa.scc.mcs":       "mcs",
+			},
+		},
+	})
+
+	restricted, err := getRestrictedSCC()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	logrus.SetLevel(logrus.DebugLevel)
+	logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+
+	ac := &admissionController{
+		client:     client,
+		restricted: restricted,
+		log:        logrus.NewEntry(logrus.StandardLogger()),
+	}
+
+	pod, err := json.Marshal(&corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Image: "regularimage",
+				},
+			},
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	req, err := json.Marshal(&admissionv1beta1.AdmissionReview{
+		Request: &admissionv1beta1.AdmissionRequest{
+			UID:      "uid",
+			Kind:     metav1.GroupVersionKind{Version: "v1", Kind: "Pod"},
+			Resource: metav1.GroupVersionResource{Version: "v1", Resource: "pods"},
+			Object: runtime.RawExtension{
+				Raw: pod,
+			},
+		}})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	reqNoUID, err := json.Marshal(&admissionv1beta1.AdmissionReview{
```

Comment:
what is any of this actually testing?

Reply:
Malformed review requests. We have a separate code path for review requests with empty UID which leads to an error. I don't know if testing it is necessary.